File: frontend/src/types/applicationState.ts
import type { AppInstanceView, AppRuntimeView } from './app';
import type { AppOwnershipView } from './appOwnership';
import type { ObservedServiceView } from './observedService';

export type ApplicationState = {
  managedApps: AppInstanceView[];
  runtimeApps: AppRuntimeView[];
  observedServices: ObservedServiceView[];
  pinnedExternalServices: ObservedServiceView[];
  foundServices: ObservedServiceView[];
  ownershipViews: AppOwnershipView[];
  updatedAt: string;
  stale?: boolean;
  refreshStatus?: 'idle' | 'running' | 'failed' | string;
  refreshStartedAt?: string | null;
  refreshCompletedAt?: string | null;
  nextRefreshAt?: string | null;
  lastError?: string | null;
};

File: backend/src/main/java/com/projectos/apps/ApplicationState.java
package com.projectos.apps;

import java.time.Instant;
import java.util.List;

import com.projectos.host.ObservedServiceView;
import com.projectos.marketplace.install.AppInstanceView;
import com.projectos.marketplace.install.AppRuntimeView;

public record ApplicationState(
        List<AppInstanceView> managedApps,
        List<AppRuntimeView> runtimeApps,
        List<ObservedServiceView> observedServices,
        List<ObservedServiceView> pinnedExternalServices,
        List<ObservedServiceView> foundServices,
        List<AppOwnershipView> ownershipViews,
        Instant updatedAt) {
}
